ambry-cloud/src/main/java/com.github.ambry.cloud/azure/AzureCloudDestination.java

```diff
@@ -36,38 +48,68 @@
 /**
  * Implementation of {@link CloudDestination} that interacts with Azure Blob Storage service.
  */
+@SuppressWarnings({"ALL", "MagicConstant"})
 class AzureCloudDestination implements CloudDestination {
 
   private static final Logger LOGGER = LoggerFactory.getLogger(AzureCloudDestination.class);
   private final CloudStorageAccount azureAccount;
   private final CloudBlobClient azureBlobClient;
+  private final DocumentClient documentClient;
+  private final String cosmosCollectionLink; // eg "/dbs/ambry-metadata/colls/blob-metadata"
+  private final RequestOptions defaultRequestOptions = new RequestOptions();
+
+  public static final String STORAGE_CONFIG_SPEC = "storageConfigSpec";
+  public static final String COSMOS_ENDPOINT = "cosmosEndpoint";
+  public static final String COSMOS_COLLECTION_LINK = "cosmosCollectionLink";
+  public static final String COSMOS_KEY = "cosmosKey";
 
   /**
-   * Construct an Azure cloud destination from a container's config spec.
-   * @param configSpec the config spec to use.
+   * Construct an Azure cloud destination from config properties.
+   * @param verProps the {@link VerifiableProperties} to use.
    * @throws InvalidKeyException if credentials in the connection string contain an invalid key.
    * @throws URISyntaxException if the connection string specifies an invalid URI.
    */
-  AzureCloudDestination(String configSpec) throws URISyntaxException, InvalidKeyException {
-    this(CloudStorageAccount.parse(configSpec));
+  AzureCloudDestination(VerifiableProperties verProps) throws URISyntaxException, InvalidKeyException {
```

Comment:
I created `CloudConfig` file in [PR#1134](https://github.com/linkedin/ambry/pull/1134/files#diff-d3379d14ef977b826aa2cae96a767822). We can use it as a centralized place for cloud related configs.

Reply:
Will change in a follow up PR.